### DISCUSSION TITLE
Sending VP9 from ffmpeg

### DISCUSSION THREAD

**Discussion by ZhaoruiGit** [paullouisageneau/libdatachannel#917] [General]
hi, all,

I have VP9 encoded frames (generated by ffmpeg), and want to stream them to the webbrowser on another PC. So very similar to the streamer example, except that VP9 codec is used for the video. 

Before I go further with the next steps, could you help me to double check the next:,
-   I need to write my own VP9packetizer, similar to [this one](https://github.com/paullouisageneau/libdatachannel/blob/master/include/rtc/h264rtppacketizer.hpp), right?
-  To write a VP9packetizer, [this](https://chromium.googlesource.com/external/webrtc/+/a9455ab235e1169572f9eae3175cd9310d6729e2/webrtc/modules/rtp_rtcp/source/rtp_format_vp9.cc) is a good example to follow, right?

Thanks a lot in advance!
Zhaorui

**Comment by paullouisageneau** [comment]
Hi, 

Yes, you are correct! 

Also, someone submitted a draft VP8 packetizer some time ago, it might be useful to you: https://github.com/paullouisageneau/libdatachannel/issues/558#issuecomment-1087985814

**Reply by ZhaoruiGit** [reply]
Thanks a lot! The example is very useful. By re-using most of the code, and filling in valide VP9 header information, I can send VP9 packets to the other PC. Really great help!

Also, if I want to use VP9 profile 2, does the next call provide correct profile string?

addVideoCodec(payloadType, "VP9", **"profile-id=2"**);
